STEER/STEER/AliGenerator.h

```diff
@@ -89,28 +93,32 @@ class AliGenerator : public TNamed, public AliRndm
 	{ox=fOrigin.At(0);oy=fOrigin.At(1);oz=fOrigin.At(2);}
     virtual void GetOrigin(TLorentzVector &o) const
 	{o[0]=fOrigin.At(0);o[1]=fOrigin.At(1);o[2]=fOrigin.At(2);o[3]=0;}
-  // Stack 
+  // Stack
     void SetStack (AliStack *stack) {fStack = stack;}
     AliStack* GetStack(){return fStack;}
   // Collision Geometry
     virtual Bool_t ProvidesCollisionGeometry() const {return kFALSE;}
-    virtual Bool_t NeedsCollisionGeometry()    const {return kFALSE;}    
+    virtual Bool_t NeedsCollisionGeometry()    const {return kFALSE;}
     virtual AliCollisionGeometry* CollisionGeometry() const {return fCollisionGeometry;}
     virtual void SetCollisionGeometry(AliCollisionGeometry* geom) {fCollisionGeometry = geom;}
 
     virtual Float_t GetEnergyCMS() const { return fEnergyCMS; }
     virtual void    SetEnergyCMS(Float_t energy = 0) { fEnergyCMS = energy; }
     virtual void    GetProjectile(TString& tar, Int_t& a, Int_t& z) const
-    {tar = fProjectile; a = fAProjectile; z = fZProjectile;}    
+    {tar = fProjectile; a = fAProjectile; z = fZProjectile;}
     virtual void    GetTarget(TString& tar, Int_t& a, Int_t& z) const
-    {tar = fTarget; a = fATarget; z = fZTarget;}    
+    {tar = fTarget; a = fATarget; z = fZTarget;}
     virtual void    SetProjectile(TString proj="", Int_t a = 0, Int_t z = 0)
 	{fProjectile = proj; fAProjectile = a; fZProjectile = z;}
     virtual void    SetTarget(TString tar="", Int_t a = 0, Int_t z = 0)
 	{fTarget = tar; fATarget = a; fZTarget = z;}
-    
+
     virtual void    SetSeed(UInt_t seed);
-    
+
+    ///-------------------------------------------------------------------------
+    virtual void SetUserTrigger(std::function<Bool_t(AliStack*)> fn) { fSetUserTrig = kTRUE; fUserTrigger = fn; }     // Enable user trigger
```

Comment:
Ciao @lucamicheletti93 , 
You need to add the include of AliStack.h and "#include <functional>", or it won't compile.

Reply:
I have added the missing include in AliGenerator.h  and I have committed the code!